src/CoreWCF.ConfigurationManager/src/CoreWCF/Configuration/ConfigurationManagerServiceModelOptions.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections;
+using System.Configuration;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace CoreWCF.Configuration
+{
+    internal class ConfigurationManagerServiceModelOptions : IConfigureNamedOptions<ServiceModelOptions>, IDisposable
+    {
+        private readonly Lazy<ServiceModelSectionGroup> _section;
+        private readonly WrappedConfigurationFile _file;
+
+        private readonly IConfigurationHolder _holder;
+
+        public ConfigurationManagerServiceModelOptions(IServiceProvider builder, string path, bool isOptional)
+        {
+            _holder = builder.GetRequiredService<IConfigurationHolder>();
+            _file = new WrappedConfigurationFile(path);
+
+            _section = new Lazy<ServiceModelSectionGroup>(() =>
+            {
+                System.Configuration.Configuration configuration = ConfigurationManager.OpenMappedMachineConfiguration(new ConfigurationFileMap(_file.ConfigPath));
+                var section = ServiceModelSectionGroup.GetSectionGroup(configuration);
```

Comment:
I found the api's you need to use so you don't need to merge the app.config with the machine.config.
```c#
ExeConfigurationFileMap fileMap = new ExeConfigurationFileMap
{
    ExeConfigFilename = path,
    MachineConfigFilename = machineConfigFilePath
};
System.Configuration.Configuration configuration = ConfigurationManager.OpenMappedExeConfiguration(fileMap, ConfigurationUserLevel.None);
```
This way you don't need to manually merge them.